lib/javascripts/base_app.js

```diff
@@ -68,6 +68,10 @@ function BaseApp(zafClient, data) {
   let evt = { firstLoad: true };
   resolveHandler(this, 'app.created')();
   resolveHandler(this, 'app.activated')(evt, evt);
+  $(window).unload(() => {
+    resolveHandler(this, 'app.willDestroy')();
+    delete this.events['app.willDestroy']; // make sure this event only fires once
```

Comment:
@danielbreves this was a suggestion from @svizzari to make the event only ever fire once in case we also get an event of the same name firing from the fw